subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

```diff
@@ -41,43 +41,38 @@
 #define FIXED_PARTITION_IS_RUNNING_APP_PARTITION(label)	\
 	 (FIXED_PARTITION_OFFSET(label) == CONFIG_FLASH_LOAD_OFFSET)
 
-#if FIXED_PARTITION_EXISTS(slot0_partition)
-#if FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot0_partition)
-#define NUMBER_OF_ACTIVE_IMAGE 0
-#endif
-#endif
-
-#if !defined(NUMBER_OF_ACTIVE_IMAGE) && FIXED_PARTITION_EXISTS(slot0_ns_partition)
-#if FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot0_ns_partition)
-#define NUMBER_OF_ACTIVE_IMAGE 0
-#endif
-#endif
-
-#if !defined(NUMBER_OF_ACTIVE_IMAGE) && FIXED_PARTITION_EXISTS(slot1_partition)
-#if FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot1_partition)
-#define NUMBER_OF_ACTIVE_IMAGE 0
-#endif
-#endif
-
-#if !defined(NUMBER_OF_ACTIVE_IMAGE) && FIXED_PARTITION_EXISTS(slot2_partition)
-#if FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot2_partition)
-#define NUMBER_OF_ACTIVE_IMAGE 1
-#endif
-#endif
-
-#if !defined(NUMBER_OF_ACTIVE_IMAGE) && FIXED_PARTITION_EXISTS(slot3_partition)
-#if FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot3_partition)
-#define NUMBER_OF_ACTIVE_IMAGE 1
-#endif
+BUILD_ASSERT(sizeof(struct image_header) == IMAGE_HEADER_SIZE,
+	     "struct image_header not required size");
+
+#if CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER >= 2
+#if FIXED_PARTITION_EXISTS(slot0_ns_partition) &&			\
+	FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot0_ns_partition)
+#define ACTIVE_IMAGE_IS 0
+#elif FIXED_PARTITION_EXISTS(slot0_partition) &&			\
+	FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot0_partition)
+#define ACTIVE_IMAGE_IS 0
+#elif FIXED_PARTITION_EXISTS(slot1_partition) &&			\
+	FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot1_partition)
+#define ACTIVE_IMAGE_IS 0
+#elif FIXED_PARTITION_EXISTS(slot2_partition) &&			\
+	FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot2_partition)
+#define ACTIVE_IMAGE_IS 1
+#elif FIXED_PARTITION_EXISTS(slot3_partition) &&			\
+	FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot3_partition)
+#define ACTIVE_IMAGE_IS 1
+#elif FIXED_PARTITION_EXISTS(slot4_partition) &&			\
+	FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot4_partition)
+#define ACTIVE_IMAGE_IS 2
+#elif FIXED_PARTITION_EXISTS(slot5_partition) &&			\
+	FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot5_partition)
+#define ACTIVE_IMAGE_IS 2
+#else
+#define ACTIVE_IMAGE_IS 0
```

Comment:
Just to make sure - can the `slot0` and `slot1` partitions be undefined for a single image case? If not then we may not need to distinguish the `#if CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER >= 2` case here

Reply:
There is other code that will topple over when there is no slot0 or slot1, on the other one there is conditional code for slot3-5. So the later need checks because code will compile without features.